snuba/web/rpc/v1/resolvers/R_eap_items/routing_strategies/outcomes_based.py

```diff
@@ -0,0 +1,205 @@
+import uuid
+from typing import cast
+
+from google.protobuf.json_format import MessageToDict
+from sentry_protos.snuba.v1.request_common_pb2 import RequestMeta, TraceItemType
+
+from snuba import state
+from snuba.attribution.appid import AppID
+from snuba.attribution.attribution_info import AttributionInfo
+from snuba.clickhouse.query import Expression
+from snuba.datasets.entities.entity_key import EntityKey
+from snuba.datasets.entities.factory import get_entity
+from snuba.datasets.pluggable_dataset import PluggableDataset
+from snuba.downsampled_storage_tiers import Tier
+from snuba.query import SelectedExpression
+from snuba.query.data_source.simple import Entity
+from snuba.query.dsl import Functions as f
+from snuba.query.dsl import and_cond, column, in_cond, literal, literals_array
+from snuba.query.logical import Query
+from snuba.query.query_settings import HTTPQuerySettings
+from snuba.request import Request as SnubaRequest
+from snuba.web.query import run_query
+from snuba.web.rpc.common.common import (
+    timestamp_in_range_condition,
+    treeify_or_and_conditions,
+)
+from snuba.web.rpc.v1.resolvers.R_eap_items.storage_routing.routing_strategies.storage_routing import (
+    BaseRoutingStrategy,
+    ClickhouseQuerySettings,
+    RoutingContext,
+)
+
+
+# TODO import these from sentry-relay
+class Outcome:
+    SPAN_INDEXED = 16
+    LOG_ITEM = 23
+
+
+_ITEM_TYPE_TO_OUTCOME = {
+    TraceItemType.TRACE_ITEM_TYPE_SPAN: Outcome.SPAN_INDEXED,
+    TraceItemType.TRACE_ITEM_TYPE_LOG: Outcome.LOG_ITEM,
+}
+
+
+def project_id_and_org_conditions(meta: RequestMeta) -> Expression:
+    return and_cond(
+        in_cond(
+            column("project_id"),
+            literals_array(
+                alias=None,
+                literals=[literal(pid) for pid in meta.project_ids],
+            ),
+        ),
+        f.equals(column("org_id"), meta.organization_id),
+    )
+
+
+class OutcomesBasedRoutingStrategy(BaseRoutingStrategy):
+    def get_ingested_items_for_timerange(self, routing_context: RoutingContext) -> int:
+        entity = Entity(
+            key=EntityKey("outcomes"),
+            schema=get_entity(EntityKey("outcomes")).get_data_model(),
+            sample=None,
+        )
+        query = Query(
+            from_clause=entity,
+            selected_columns=[
+                SelectedExpression(
+                    name="num_items",
+                    expression=f.sum(column("quantity"), alias="num_items"),
+                )
+            ],
+            condition=and_cond(
+                project_id_and_org_conditions(routing_context.in_msg.meta),
+                timestamp_in_range_condition(
+                    routing_context.in_msg.meta.start_timestamp.seconds,
+                    routing_context.in_msg.meta.end_timestamp.seconds,
+                ),
+                f.equals(column("outcome"), 0),
```

Comment:
We have a `Outcome.ACCEPTED` or something to replace the 0, don't we?

Reply:
done